internal/controller/vulnerabilityreport_controller.go

```diff
@@ -84,6 +84,8 @@ func (r *VulnerabilityReportReconciler) Reconcile(ctx context.Context, req ctrl.
 	scanJob.Status.ScannedImagesCount = len(vulnerabilityReports.Items)
 	if scanJob.Status.ScannedImagesCount == scanJob.Status.ImagesCount {
 		scanJob.MarkComplete(sbombasticv1alpha1.ReasonAllImagesScanned, "All images scanned successfully")
+	} else {
```

Comment:
If the scan fails during the middle of the operation (e.g., SBOM generation or scanning), would setting the status to InProgress still be accurate?

Currently, we always mark the job as InProgress when ScannedImagesCount < ImagesCount, but I think we should first check that the job has not already failed before updating the condition.

What do you think? Should we add a condition to prevent overwriting a Failed state?